server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -457,6 +457,9 @@ private void recoverFromRemoteStore(IndexShard indexShard, Repository repository
             // Download segments from remote segment store
             indexShard.syncSegmentsFromRemoteSegmentStore(true);
 
+            if (store.directory().listAll().length == 0) {
+                store.createEmpty(indexShard.indexSettings().getIndexVersionCreated().luceneVersion);
```

Comment:
Just wondering since the creation of empty directory depends on the lucene version, do we need to handle anything for version upgrade?

Reply:
It should not change anything in the version compatibility as the index already exists. I think the confusion is due to PR title. We are not creating a new OpenSearch index but empty lucene index :). Let me change the title.